Utilities.cs

```diff
@@ -149,9 +147,13 @@ public unsafe static string Utf8StringPtrToString (IntPtr utf8Str)
 			if (len == 0)
 				return String.Empty;
 
-			var bytes = new byte[len];
-			Marshal.Copy (utf8Str, bytes, 0, len);
-			return Encoding.UTF8.GetString (bytes);
+			var bytes = ArrayPool<byte>.Shared.Rent (len);
+			try {
+				Marshal.Copy (utf8Str, bytes, 0, len);
+				return Encoding.UTF8.GetString (bytes, 0, len);
```

Comment:
Note that `ArrayPool` might return a `byte[]` larger than we asked for. We need to make sure `GetString` uses `len`.